src/api/app/services/session_controller_service/session_creator.rb

```diff
@@ -0,0 +1,23 @@
+module SessionControllerService
```

Comment:
This is a lot of glue code for `User.find_with_credentials`. Are you planing to move code in from that end?

Reply:
I didn't clean my whole stash. Now it's done :)

Reply:
> I didn't clean my whole stash. 

Not sure what you mean by that, can you elaborate?

> The idea was to clean up some logic from the SessionController#create

Sure thing, this is good. But that is only half of the logic right? The rest is behind `User.find_with_credentials`, my question was if you want to tackle this too?

Reply:
> Not sure what you mean by that, can you elaborate?
> 
> >

Ah I had some changes on my stash that I didn't push, specially the changes on the controller.. may bad.

Reply:
Your question was if would make sense to move the whole find_with_credentials* to a service? Maybe, should it happen in this PR? not necessarily.

Reply:
> This is a lot of glue code for `User.find_with_credentials`. Are you planing to move code in from that end?

if you look the new `create` method it is straight forward, no more nested conditions of `if xxx; redirect and return` which in my opinion can be better handled with `before_filters`

Reply:
just look the file https://github.com/openSUSE/open-build-service/blob/7cba5648875dbfda6ccd91b868d8e5695c17610b/src/api/app/controllers/webui/session_controller.rb#L14 the method is just straight forward, easy to understand without hide important information. Just beautiful :)